### PR TITLE
Rework CI

### PR DESCRIPTION
**Pull request by nwf-msr**

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -94,7 +79,52 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Install build dependencies
+            # The homebrew packages are broken at the moment and error out
+            # after trying to install Python as a dependency of ninja because
+            # 2to3 exists.  As a quick hack, delete it first.  This should be
+            # removed once the homebrew install is fixed.
+      run: "rm -f /usr/local/bin/2to3 ; brew update && brew install ninja"
     
     - uses: mjp41/workaround8649@c8550b715ccdc17f89c8d5c28d7a48eeff9c94a8
```
Comment (mjp41) [inline .github/workflows/main.yml:129]: I don't think this is needed on Mac OS

**Review by mjp41 — APPROVED** [review]
LGTM

File: .github/workflows/main.yml (hunk 1 above)
Comment (nwf-msr) [reply .github/workflows/main.yml:129]: OK, let's give that a shot. :)

Comment (nwf-msr) [reply .github/workflows/main.yml]: Looks like you're right. :)

**Pull request merged**